src/unify/tests.rs

```diff
@@ -474,3 +488,177 @@ fn clone_table() {
         }
     }
 }
+
+enum UndoLog {
```

Comment:
Can you maybe move this to a `tests/foo.rs` file and add some comments explaining what it is trying to test? There's a lot of moving parts here. =)